### ISSUE TITLE
podman logs -f continues indefinitely on a stopped container

### ISSUE THREAD

**Issue by jgallucci32** [containers/libpod#6531]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**
/kind bug

**Description**
Running `podman logs -f` on a stopped container run indefinitely and does not exit
This is a regression of Issue https://github.com/containers/libpod/issues/435 and PR https://github.com/containers/libpod/pull/437

**Steps to reproduce the issue:**

1. Run a container which exits immediately
```
podman run -d --name hello-world hello-world
```

2. Follow logs of stopped container
```
podman logs -f hello-world
```

**Describe the results you received:**
Command does not exit; hangs indefinitely

**Describe the results you expected:**
Command should exit.

**Additional information you deem important (e.g. issue happens only occasionally):**
The expected behavior of Docker is to exit when following a log of a stopped container

**Output of `podman version`:**

```
Version:      2.0.0-dev
API Version:  1
Go Version:   go1.14.3
Git Commit:   d857275901e8c1ea7515360631e5894018e17f30
Built:        Fri Jun  5 17:00:00 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.0-dev
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.18-0.6.dev.git50aeae4.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.18-dev, commit: 51e91bbc42aaf0676bb4023fb86f00460bf7a0a2'
  cpus: 4
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: file
  hostname: gallucci-fedora.devlnk.net
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.7.0-1.fc33.x86_64
  linkmode: dynamic
  memFree: 15239475200
  memTotal: 16783962112
  ociRuntime:
    name: crun
    package: crun-0.13-2.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 4294963200
  swapTotal: 4294963200
  uptime: 22h 20m 14.55s (Approximately 0.92 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 6
    paused: 0
    running: 4
    stopped: 2
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 3
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 1
  Built: 1591401600
  BuiltTime: Fri Jun  5 17:00:00 2020
  GitCommit: d857275901e8c1ea7515360631e5894018e17f30
  GoVersion: go1.14.3
  OsArch: linux/amd64
  Version: 2.0.0-dev
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-2.0.0-0.121.dev.git1fcb678.fc33.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Fedora 33 (Rawhide Prerelease)

**Comment by jgallucci32** [comment]
Did some extended testing and confirm the issue exists in older versions of podman in addition to the one reported:
* podman 1.6.4 - Hangs
* podman 1.9.3 - Hangs

**Comment by rhatdan** [comment]
Just for grins, what does Docker do?

**Comment by jgallucci32** [comment]
Docker exits properly. Podman used to have this same functionality and was
fixed in a bug 2 years ago, but the log.go code was completely rewritten so
it is a regression.

**Comment by rhatdan** [comment]
Great thanks.

**Comment by rhatdan** [comment]
@jgallucci32 Interested on working on a fix and opening a PR?

**Comment by jgallucci32** [comment]
/reopen

This issue was originally fixed in #6591 but has been reverted in #6644 to address problems with tests failing.
